src/engine/WindowManager.cxx

```diff
@@ -198,9 +215,15 @@ void WindowManager::setScreenResolution(int mode)
 
 void WindowManager::renderScreen()
 {
+  auto *renderer = WindowManager::instance().getRenderer();
```

Comment:
Why did you make this a variable and change `WindowManager::instance().getRenderer()` to `renderer`? When making this many changes to the codebase like replacing the UI with a new framework, I feel like the less unnecessary changes the better.

Reply:
look here, in instance function you have condition and mutex lock/unlock operations
so below you have 2 conditions and 4 mutex operations when it need only one
just using better flow in render call